File: levenshtein.py
import sys

if len(sys.argv) != 3:
    raise TypeError('Function requries exactly two arguments. You supplied {}'.format(len(sys.argv) - 1))

w1, w2 = sys.argv[1].lower(), sys.argv[2].lower()

def lev(word1, word2):
    m, n = len(word1) + 1, len(word2) + 1
    v = [None] * (n)
    for i in range(n):
        v[i] = [None] * (m)
    v[0] = list(range(m))
    for i in range(n):
        v[i][0] = i

    for i in range(1, n):
        for j in range(1, m):
            v[i][j] = min(
                v[i-1][j] + 1,
                v[i][j-1] + 1,
                v[i-1][j-1] + 2 * (1 if word1[j-1]!=word2[i-1] else 0)
            )

    for i in v:
        print(i)

    return v[n-1][m-1]

print(lev(w1, w2))

File: levenshtein.c
#include <stdio.h>

int lev(char w1[], char w2[]) {
    printf("word1: %s\n", w1);
    printf("word2: %s\n", w2);
    return 5;
}

int main(int argc, char *argv[]) {
    if (argc != 3) {
        // TODO: use errors/exceptions to handle bad input
        printf("Function requires exactly two arguments, you provided %d\n", argc - 1);
        return 1;
    }

    printf("Levenshtein distance between %s and %s is %d\n", argv[1], argv[2], lev(argv[1], argv[2]));

    return 1;
}

